src/devices/SM28BYJ48/SM28BYJ48.cs

```diff
@@ -0,0 +1,202 @@
+        /// <returns>Current stepper motor.</returns>
+        public SM28BYJ48 SetStepperDelay(int ms)
+        {
+            _delayMs = ms > 1 ? ms : 1;
+            return this;
+        }
+
+        /// <summary>
+        /// This method is being called when the motor has completed the amount of steps.
+        /// </summary>
+        /// <param name="movementCompleted">The callback.</param>
+        /// <returns>Current stepper motor.</returns>
+        public SM28BYJ48 AddMovementCompletionHandler(Action movementCompleted)
+        {
+            _movementCompleted = new Action(movementCompleted);
+            return this;
+        }
+
+        /// <summary>
+        /// Stop the motor.
+        /// </summary>
+        public void Stop()
+        {
+            _rotationDirection = RotationDirection.Stopped;
+            _controller.Write(_pin1, PinValue.Low);
+            _controller.Write(_pin2, PinValue.Low);
+            _controller.Write(_pin3, PinValue.Low);
+            _controller.Write(_pin4, PinValue.Low);
+        }
+
+        /// <summary>
+        /// Rotate the motor clockwise.
+        /// </summary>
+        /// <param name="steps">Number of steps.</param>
+        public void RotateClockwise(int steps)
+        {
+            SetRotationDirection(RotationDirection.Clockwise, steps);
+            Start();
+        }
+
+        /// <summary>
+        /// Rotate the motor counter clockwise.
+        /// </summary>
+        /// <param name="steps">Number of steps.</param>
+        public void RotateCounterClockwise(int steps)
+        {
+            SetRotationDirection(RotationDirection.CounterClockwise, steps);
+            Start();
+        }
+
+        private void SetRotationDirection(RotationDirection direction, int steps)
+        {
+            _rotationDirection = direction;
+            _steps = steps;
+            _currentStep = 0;
+        }
+
+        private void Start()
+        {
+            if (_steps < 0)
+                _rotationDirection = RotationDirection.Stopped;
+
+            while (_rotationDirection != RotationDirection.Stopped)
+            {
+                Thread.Sleep(_delayMs);
+
+                if (_rotationDirection == RotationDirection.Clockwise)
+                    _engineStep = _engineStep - 1 < 1 ? 8 : _engineStep - 1;
+                else
+                    _engineStep = _engineStep + 1 > 8 ? 1 : _engineStep + 1;
+
+                ApplyEngineStep();
+                _currentStep++;
+
+                if (_currentStep >= _steps)
+                {
+                    _rotationDirection = RotationDirection.Stopped;
+                    _movementCompleted?.Invoke();
+                }
+            }
+        }
+
+        private void ApplyEngineStep()
+        {
+            if (_type == StepperType.HalfStep)
+            {
+                _controller.Write(_pin1, _engineStep == 8 || _engineStep == 1 || _engineStep == 2 ? PinValue.High : PinValue.Low);
```

Comment:
you can remove `? PinValue.High : PinValue.Low` now, we accept bool values since few days